### PR TITLE
Fix HTTP 308 #263

### PR DESCRIPTION
**Pull request by FredLL-Avaiga**
- impact on https via reverse proxy
- change: / not ending page routes

### CODE REVIEW

**Comment by florian-vuillemot** [comment]
So wonderful !

File: taipy/gui/server.py [hunk 1]
```diff
@@ -154,22 +153,22 @@ def _render(self, html_fragment, style, head):
     def _direct_render_json(self, data):
         return jsonify(data)
 
-    def _render_page(self) -> t.Any:
```
Comment (dinhlongviolin1) [inline taipy/gui/server.py:157]: Nice one :D

File: taipy/gui/gui.py [hunk 2]
```diff
@@ -1170,13 +1170,10 @@ def run(self, run_server: bool = True, run_in_thread: bool = False, ssl_context:
         )
 
         # Run parse markdown to force variables binding at runtime
-        # (save rendered html to page.rendered_jsx for optimization)
-        for page in self._config.pages + self._config.partials:  # type: ignore
-            # Server URL Rule for each page jsx
-            pages_bp.add_url_rule(f"/taipy-jsx/{page._route}/", view_func=self._server._render_page)
+        pages_bp.add_url_rule("/taipy-jsx/<path:page_name>", view_func=self._server._render_page)
```
Comment (dinhlongviolin1) [inline taipy/gui/gui.py:1173]: Good fix!

File: gui/src/components/Router.tsx [hunk 3]
```diff
@@ -55,7 +55,7 @@ const Router = () => {
         }
         // Fetch Flask Rendered JSX React Router
         axios
-            .get<AxiosRouter>(`/taipy-init/?client_id=${state.id || ""}`)
+            .get<AxiosRouter>("/taipy-init", {params: {client_id:state.id || ""}})
```
Comment (dinhlongviolin1) [inline gui/src/components/Router.tsx:58]: I guess this is the problem that this PR is trying to fix right

**Review by dinhlongviolin1 — APPROVED** [review]
It works now which is great

**Review by FabienLelaquais — APPROVED** [review]
👌

**Comment by florian-vuillemot** [comment]
Will we merge it in taipy-gui 1.0 ?

**Comment by FredLL-Avaiga** [comment]
Only if necessary or you want it ;-)

**Pull request merged**